onnxruntime/core/providers/coreml/builders/impl/argmax_op_builder.cc

```diff
@@ -32,57 +32,40 @@ Status ArgMaxOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder,
 
   NodeAttrHelper helper(node);
   const auto axis = helper.Get("axis", 0);
-  const auto keep_dims = helper.Get("keep_dims", 1);
-  const bool removedim = keep_dims != 1;
+  const auto keepdims = helper.Get("keepdims", 1);
+  const bool removedim = keepdims != 1;
 
   auto* coreml_argmax = layer->mutable_argmax();
   coreml_argmax->set_axis(axis);
   coreml_argmax->set_removedim(removedim);
 
-  // Get ArgMax's next node(Cast)'s outputdefs
-  auto it = node.OutputEdgesBegin();
-  const auto* succ_node(graph_viewer.GetNode(it->GetNode().Index()));
+  // There are two cases here:
+  // 1. Special Case (ArgMax-Cast), we fuse the Argmax's output/Cast's input
+  // (We still have this special case here because CoreML model does not have Cast)
+  // 2. Otherwise, we add Argmax layer normally
+  if (node.GetOutputEdgesCount() == 1) {
+    auto it = node.OutputEdgesBegin();
+    const auto* succ_node(graph_viewer.GetNode(it->GetNode().Index()));
+    if (succ_node->OpType() == "Cast") {
+      // Skip the cast's input/argmax's output
```

Comment:
Is it just that there's a Cast, or that it's specifically a Cast from int64 to int32? If it's the latter, and that's checked somewhere else, there should be a comment here explaining why we don't need to check the to/from of the Cast